tests/demonstrator_scripts/test_computation_volume_content.py

```diff
@@ -0,0 +1,138 @@
+from pint.testsuite.helpers import assert_quantity_almost_equal as assert_approx
+from lebedigital.demonstrator_scripts.computation_volume_content import computation_volume_content
+from lebedigital.demonstrator_scripts.computation_volume_content import computation_ratios
+from lebedigital.unit_registry import ureg
+
+def test_computation_volume_content() :
+    """
+    This is the test for the computation_volume_content function.
+    It test all output with  and performs a number of sanity checks
+    """
+
+    # TEST No 1, check intended use in general
+    input1 = {}
+    # densities
+    input1['density_cem'] = 1.44 * ureg('g/cm^3')
+    input1['density_sub'] = 840 * ureg('kg/m^3')
+    input1['density_water'] = 977 * ureg('kg/m^3')
+    input1['density_plasticizer'] = 0.98 * ureg('g/cm^3')
+    input1['density_aggregates'] = 1.5 * ureg('t/m^3')
+
+    input1['wb_mass_ratio'] = 0.4 * ureg('dimensionless')
+    input1['sc_volume_fraction'] = 0.4 * ureg('dimensionless')
+    input1['aggregates_volume_fraction'] = 0.65 * ureg('dimensionless')
+    input1['plasticizer_volume_content'] = 10 * ureg('kg/m^3')
+
+    output1 = computation_volume_content(input1)
+
+    # check all values
+    assert_approx(output1['water_vol_fraction'], 0.10510134 * ureg('dimensionless'), rtol=1e-6)
+    assert_approx(output1['sub_vol_fraction'], 0.0938778312 * ureg('dimensionless'), rtol=1e-6)
+    assert_approx(output1['cem_vol_fraction'], 0.140816747 * ureg('dimensionless'), rtol=1e-6)
+    assert_approx(output1['cem_vol_content'], 202.776115 * ureg('kg/m^3'), rtol=1e-6)
+    assert_approx(output1['sub_vol_content'], 78.8573782 * ureg('kg/m^3'), rtol=1e-6)
+    assert_approx(output1['water_vol_content'], 102.68401 * ureg('kg/m^3'), rtol=1e-6)
+    assert_approx(output1['aggregates_vol_content'], 975.0 * ureg('kg/m^3'), rtol=1e-6)
+    assert_approx(output1['density_paste'], 1126.62144 * ureg('kg/m^3'), rtol=1e-6)
```

Comment:
Could it happen, that you input values that lead to the sanity checks performed in the library to fail? If so, such a test should be added here.

Reply:
I do not think so. I implemented them to be sure everything was "sane", while I was hunting some other bug. I decided I might as well leaven them there, in case for future changes.
I have not tried "stupid input", i.e. negative densities or something strange.
Assuming physically reasonable input, I think this is a robust function.